MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm

```diff
@@ -408,15 +415,33 @@ static inline CIE1931XY VkXYColorEXTToCIE1931XY(VkXYColorEXT xy) {
 VkResult MVKSwapchain::getRefreshCycleDuration(VkRefreshCycleDurationGOOGLE *pRefreshCycleDuration) {
 	if (_device->getConfigurationResult() != VK_SUCCESS) { return _device->getConfigurationResult(); }
 
+#if MVK_IOS_OR_TVOS || MVK_MACCAT
 	NSInteger framesPerSecond = 60;
-#if MVK_IOS_OR_TVOS
 	UIScreen* screen = [UIScreen mainScreen];
 	if ([screen respondsToSelector: @selector(maximumFramesPerSecond)]) {
 		framesPerSecond = screen.maximumFramesPerSecond;
 	}
 #endif
-#if MVK_MACOS
-	// TODO: hook this up for macOS, probably need to use CGDisplayModeGetRefeshRate
+
+#if MVK_MACOS && !MVK_MACCAT
+	// Find the screen for the window whose content view is backed by the swapchains CAMetalLayer.
+	// Default to the mainScreen if no such window can be found.
+	NSScreen *screen = [NSScreen mainScreen];
+	for (NSWindow* window in [[NSApplication sharedApplication] windows]) {
+		NSView *view = [window contentView];
+		if (view && ([view layer] == _mtlLayer)) {
+			screen = [window screen];
+		}
+	}
```

Comment:
Hmm. So we have two flawed solutions. I personally don't have any need for this function for my project at all. I just saw the todo comment when i was working on the other bits of VK_GOOGLE_display_timing which do matter to me, and i thought this would be easy to fix, just a small loose end, so why not spend a couple of minutes? Minutes turned into hours when i realized there wasn't an easy way to get from the _mtlLayer to the NScreen. I have already spent more time on it than what i can spare, and Cocoa and all those views and layer mechanisms are not my area of expertise or experience at all.

Iow., i can't really spend more time on this low priority pull request, and the question is what to do - what is the least flawed option?

- Use my code as is?
- Replace it by your suggestion?
- Just drop the pull request and leave things as they are?

Thanks for the feedback.

Reply:
> enumerate windows and their views like you are now, and check if the view's layer either is _mtlLayer or has _mtlLayer as a direct or indirect sublayer

This sounds like it makes the best solution...albeit brute force...but it is what it is. For almost all simple cases, it should be a straightforward linear depth search...so probably not as expensive as imagined.

I'm wondering if this value might be cached once retrieved? I guess the danger is the app changing the plumbing on the fly without telling us? Or the window holding the layer moving to a different screen via the user?

> i can't really spend more time on this low priority pull request

No worries. Thanks for the input so far! We're well on our way as a result. I've marked this WIP now, and one of us can come back to it to finish it up.

Reply:
What is keeping us from combining @kleinerm and @cdavis5e's suggestions? The only issue I saw mentioned was this:
> But even this will fail with a layer-hosting view (i.e. where the view's layer property was assigned instead of letting AppKit create the layer), because AppKit doesn't set the layer's delegate to the view in that case.

We could simply avoid using the layer's delegate and resort back to the original implementation in which the window's contentView was used. So... what about this?
```objc
    CALayer* layer = _mtlLayer;
	while (layer.superlayer) {
		layer = layer.superlayer;
	}
	
	for (NSWindow* window in [[NSApplication sharedApplication] windows]) {
 		NSView *view = [window contentView];
 		if (view && ([view layer] == layer)) { // Check against layer and not _mtlLayer.
 			screen = [window screen];
 		}
 	}
```

Reply:
Probably nothing, if you'd like to contribute that bit.

Reply:
@kleinerm Shall I open a PR on your fork with the changes? I haven't tested that snippet but it seems only logical for it to work imo.